v2/test/e2e/mpi_job_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021 The Kubeflow Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+
+	common "github.com/kubeflow/common/pkg/apis/common/v1"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kubeflow "github.com/kubeflow/mpi-operator/v2/pkg/apis/kubeflow/v2beta1"
+)
+
+var _ = ginkgo.Describe("MPIJob", func() {
+	var (
+		namespace string
+		mpiJob    *kubeflow.MPIJob
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+		}
+		var err error
+		ns, err = k8sClient.CoreV1().Namespaces().Create(context.Background(), ns, metav1.CreateOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		namespace = ns.Name
+	})
+
+	ginkgo.AfterEach(func() {
+		if namespace != "" {
```

Comment:
Can it be empty?

Reply:
I see, so "Expect(err).ToNot(HaveOccurred())" doesn't stop the test from executing?

Then I think it's worth to break the test execution when it clearly doesn't make sense anymore, for example when the MPI Job's creation fails. Even when namespace creation fails, continuation gives little hope for any useful information.

Reply:
It does stop the test from executing, but `AfterEach` will still be called.